modules/reactor/src/meshgenerators/HexIDPatternedMeshGenerator.C

```diff
@@ -0,0 +1,217 @@
+#include "PatternedHexMeshGenerator.h"
+#include "HexIDPatternedMeshGenerator.h"
+
+registerMooseObject("ReactorApp", HexIDPatternedMeshGenerator);
+
+defineLegacyParams(HexIDPatternedMeshGenerator);
+
+InputParameters
+HexIDPatternedMeshGenerator::validParams()
```

Comment:
There is a significant amount of duplication between this and the Cartesian equivalent. We need to get around this

Reply:
@loganharbour 
Perhaps, I can create a new class `IDPatternedMeshGeneratorBase` which contains the duplicated functions.